---
version: "3.9"
services:
  frontend:
    build: ./frontend
    ports:
      - "5173:5173"
    volumes:
      - ./frontend:/app
      - /app/node_modules
    command: npm run dev
    environment:
      - VITE_API_BASE_URL
      - VITE_KEYCLOAK_BASE_URL
      - VITE_REALM_NAME
      - VITE_FRONTEND_CLIENT_ID
      - VITE_FRONTEND_CLIENT_REDIRECT
      - VITE_API_BASE_URL
    networks:
      - app

  node-api:
    build: ./node-api
    ports:
      - "3000:3000"
      - "3002:3002"
    volumes:
      - ./node-api/src:/app/src
      - ./node-api/drizzle:/app/drizzle
      - ./node-api/drizzle.config.ts:/app/drizzle.config.ts
      - ./node-api/tsconfig.json:/app/tsconfig.json

    depends_on:
      - postgres
      - keycloak
    restart: on-failure:10
    environment:
      - DATABASE_URL
      - DOMAIN
      - REALM_NAME
      - KEYCLOAK_BASE_URL
      - KEYCLOAK_BASE_URL
      - BACKEND_CLIENT_ID
      - MINIO_ROOT_USER
      - MINIO_ROOT_PASSWORD
      - MINIO_DATASET_BUCKET
      - MINIO_RESULTS_BUCKET
      - MINIO_HOST
      - MINIO_PORT
    networks:
      - app

  python-api:
    build: ./python-api
    ports:
      - "3001:3001"
    volumes:
      - ./python-api:/app
    command: uvicorn main:app --reload --host 0.0.0.0 --port 3001
    environment:
      - GOOGLE_EARTH_SERVICE_ACCOUNT
    networks:
      - app

  caddy:
    image: caddy:latest
    ports:
      - "80:80"
    volumes:
      - ./Caddyfile:/etc/caddy/Caddyfile
    networks:
      - app

networks:
  app:
    driver: bridge
